src/main/java/com/revealprecision/revealserver/service/BusinessStatusService.java

```diff
@@ -0,0 +1,103 @@
+package com.revealprecision.revealserver.service;
+
+import static com.revealprecision.revealserver.props.BusinessStatusProperties.GENERAL;
+
+import com.revealprecision.revealserver.enums.LookupEntityTypeTableEnum;
+import com.revealprecision.revealserver.exceptions.NotFoundException;
+import com.revealprecision.revealserver.persistence.domain.EntityTag;
+import com.revealprecision.revealserver.persistence.domain.EntityTag.Fields;
+import com.revealprecision.revealserver.persistence.domain.Task;
+import com.revealprecision.revealserver.props.BusinessStatusProperties;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.util.Pair;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class BusinessStatusService {
+
+  private final BusinessStatusProperties businessStatusProperties;
+  private final MetaDataJdbcService metaDataJdbcService;
+  private final EntityTagService entityTagService;
+
+  public void setBusinessStatus(Task task, String businessStatus) {
+
+    UUID planIdentifier = task.getAction().getGoal().getPlan().getIdentifier();
```

Comment:
has the design been updated to to EAGER fetching of these items?

Reply:
The design allows upwards not downwards so this should be fine..